### DISCUSSION TITLE
Is there a way to template "<%  %>"

### DISCUSSION THREAD

**Discussion by Archie-2021** [SilentVoid13/Templater#641] [Help] [answered]
When I use <% <% tp.file.cursor(1) %> %> it gives me an error because it execute the empty syntax. If I use \<\% <% tp.file.cursor(1) %> \%\> to supposedly escape it it puts the added supposed escape characters too and don't work, what should I do to make a template for "<%  %>"?

**Comment by welpdx** [comment, marked as answer]
Can you let us know what you want to do?

Anyways, if I am correct in assuming that you want to output `<% tp.file.cursor(1) %>`, try this:
```
<%* tR+= `<% tp.file.cursor(1) %>` %>
```
Basically, enclose the code you want to output in backticks `` ` ``.

But if this is not what you want, can you please elaborate on what you want to do? 

Hope that helps

~Welp

**Reply by Archie-2021** [reply]
> Can you let us know what you want to do?
> 
> Anyways, if I am correct in assuming that you want to output `<% tp.file.cursor(1) %>`, try this:
> 
> ```
> <%* tR+= `<% tp.file.cursor(1) %>` %>
> ```
> 
> Basically, enclose the code you want to output in backticks `` ` ``.
> 
> But if this is not what you want, can you please elaborate on what you want to do?
> 
> Hope that helps
> 
> ~Welp

What I want is to use Templater to insert `<%  %>` quickly (by a hotkey) in the text and put the cursor in the middle, basically to write Templater code faster. Like how there is options for auto pairing `"` , `[[]]`, etc in obsidian settings. I hope I could explain it properly.

**Reply by welpdx** [reply]
Thanks for elaborating. Try this:
```js
<%*
if (tp.file.selection() !== "" ) {
 tR += `<% ` + tp.file.selection() + ` %>`
}
%>
```
Select text and then run this script via Open Insert Template Modal.

**Reply by Archie-2021** [reply]
> Thanks for elaborating. Try this:
> 
> ```js
> <%*
> if (tp.file.selection() !== "" ) {
>  tR += `<% ` + tp.file.selection() + ` %>`
> }
> %>
> ```
> 
> Select text and then run this script via Open Insert Template Modal.

Thank you. This works like a charm, altho I think you missed half of it, is there a reason for it or you just forgot?
```
<%*
if (tp.file.selection() !== "" ) {
 tR += `<% ` + tp.file.selection() + ` %>`
}
if (tp.file.selection() == "" ) {
 tR += `<% ` + tp.file.cursor(1) + ` %>`
}
%>
```

**Reply by welpdx** [reply]
> altho I think you missed half of it, is there a reason for it or you just forgot?

Ah yes, I forgot. I realized that I could use this too and started to work on a different script half way though lol. My bad. 

```js
<%*
if (tp.file.selection() !== "" ) {
 tR += `<% ` + tp.file.selection() + ` %>`
} else if (tp.file.selection() == "" ) {
 tR += `<% ` + tp.file.cursor(1) + ` %>`
} 
%>
```

> Thank you. This works like a charm
Awesome! Great question and suggestion!

---

By the way, does this answer #620?
